app/assets/javascripts/components/widgets_box.js.jsx

```diff
@@ -44,14 +44,14 @@ var Widgets = React.createClass({
   }
 })
 
-var Widget = React.createClass({
+var WidgetActions = React.createClass({
   getInitialState() {
     return { edit: false };
   },
 
   handleEdit(e){
     e.preventDefault();
-    console.log('edit');
+    this.props.toggleEditShow();
   },
 
   handleDelete(e){
```

Comment:
handleDelete is where the 'code reuse by composition not inheritance' thing starts to seem weird to me. I want all the widgets to have this delete method, but I don't understand how I can get it to work properly.